### PR TITLE
Drop database support

### PR DESCRIPTION
**Pull request by zachmu**
Also improved support for creating and dropping dbs when using the --multi-db-dir flag

### CODE REVIEW

File: integration-tests/bats/sql-create-database.bats [hunk 1]
```diff
@@ -8,94 +8,185 @@ teardown() {
...
+    dolt sql -q "drop database mydb"
+
+    [ ! -d mydb ]
 }
 
-@test "sql-create-database: create database that already exists throws an error" {
-    run dolt sql << SQL
-CREATE DATABASE mydb;
-CREATE DATABASE mydb;
+@test "sql-create-database: with multi-db-dir" {
```
Comment (max-hoffman) [inline integration-tests/bats/sql-create-database.bats:55]: this feels like 4 tests

File: integration-tests/bats/sql-server.bats [hunk 2]
```diff
@@ -1114,11 +1113,60 @@ while True:
...
+
+    mkdir no_dolt && cd no_dolt
+    mkdir db_dir
+    start_sql_server_with_args --host 0.0.0.0 --user dolt --multi-db-dir=db_dir
+
+    server_query "" 1 "create database test1"
+    server_query "" 1 "show databases" "Database\ninformation_schema\ntest1"
+    server_query "test1" 1 "create table a(x int)"
+    server_query "test1" 1 "insert into a values (1), (2)"
+    # not bothering to check the results of the commit here
```
Comment (max-hoffman) [inline integration-tests/bats/sql-server.bats:1144]: is this preferred or multi_query, which doesn't expect a result?

File: go/cmd/dolt/commands/sqlserver/server.go [hunk 3]
```diff
@@ -99,14 +105,37 @@ func Serve(ctx context.Context, version string, serverConfig ServerConfig, serve
 	dbNamesAndPaths := serverConfig.DatabaseNamesAndPaths()
 
 	if len(dbNamesAndPaths) == 0 {
-		var err error
-		mrEnv, err = env.DoltEnvAsMultiEnv(ctx, dEnv)
-		if err != nil {
-			return err, nil
+		if len(serverConfig.DataDir()) > 0 && serverConfig.DataDir() != "." {
```
Comment (max-hoffman) [inline go/cmd/dolt/commands/sqlserver/server.go:108]: i'm not sure i understand why the logic is different between the two outer cases. A noop `fs.WithWorkingDir` that just sets it to "." seems fine? If we call `MultiEnvForDirectory` when we could have called `DoltEnvAsMultiEnv` that also seems ok?

**Review by max-hoffman — COMMENTED** [review]
looks good overall but small comments; probably leave the sub dEnvs until we want to do another config refactor pass?

File: integration-tests/bats/sql-server.bats (hunk 2 above)
Comment (zachmu) [reply integration-tests/bats/sql-server.bats:1144]: I prefer it

File: go/cmd/dolt/commands/sqlserver/server.go (hunk 3 above)
Comment (zachmu) [reply go/cmd/dolt/commands/sqlserver/server.go:108]: DoltEnvAsMultiEnv includes the dolt DB in the current working directory, as well as any in subdirectories of it. It's a subtle difference, but specifying data_dir = "." should exclude the dolt DB in the current working directory.

DoltEnvAsMultiEnv is really only there to support quick start, it makes some compromises in the data model.

**Review by zachmu — COMMENTED** [review]
Thanks for feedback

**Pull request merged**